feature-layer-update-attributes/src/main/java/com/esri/arcgisruntime/sample/featurelayerupdateattributes/MainActivity.java

```diff
@@ -16,14 +16,135 @@
+
+    // set an on touch listener to listen for click events
+    mMapView.setOnTouchListener(new DefaultMapViewOnTouchListener(this, mMapView) {
+      @Override
+      public boolean onSingleTapConfirmed(MotionEvent e) {
+
+        // get the point that was clicked and convert it to a point in map coordinates
+        final Point clickPoint = mMapView.screenToLocation(new android.graphics.Point((int) e.getX(), (int) e.getY()));
+        int tolerance = 5;
+        double mapTolerance = tolerance * mMapView.getUnitsPerPixel();
+
+        // create objects required to do a selection with a query
+        Envelope envelope = new Envelope(clickPoint.getX() - mapTolerance, clickPoint.getY() - mapTolerance, clickPoint.getX() + mapTolerance, clickPoint.getY() + mapTolerance, map.getSpatialReference());
+        QueryParameters query = new QueryParameters();
+        query.setGeometry(envelope);
+        query.setMaxFeatures(1);
+
+        // call select features
+        final ListenableFuture<FeatureQueryResult> future = featureLayer.selectFeatures(query, FeatureLayer.SelectionMode.NEW);
+        // add done loading listener to fire when the selection returns
+        future.addDoneListener(new Runnable() {
+          @Override
+          public void run() {
+            try {
+              //call get on the future to get the result
+              FeatureQueryResult result = future.get();
+
+              String title = null;
+
+              //find out how many items there are in the result
+              int i = 0;
+              for (; result.iterator().hasNext(); ++i) {
+                title = (String) result.iterator().next().getAttributes().get("typdamage");
+              }
+              Toast.makeText(getApplicationContext(), i + " features selected", Toast.LENGTH_SHORT).show();
+
+              // create a textview for the callout
+              RelativeLayout calloutLayout = new RelativeLayout(getApplicationContext());
+              RelativeLayout.LayoutParams relativeParams = new RelativeLayout.LayoutParams(
+                  RelativeLayout.LayoutParams.MATCH_PARENT, RelativeLayout.LayoutParams.MATCH_PARENT);
+              relativeParams.addRule(RelativeLayout.ALIGN_PARENT_TOP);
+
+              TextView calloutContent = new TextView(getApplicationContext());
+              calloutContent.setId(R.id.textview);
+              calloutContent.setTextColor(Color.BLACK);
+              calloutContent.setSingleLine();
+              calloutContent.setText(title);
+//              calloutContent.setText("X:" +  (String.format("%.2f", clickPoint.getX()))
+//                  + ", y:" + (String.format("%.2f", clickPoint.getY())));
+
+              relativeParams.addRule(RelativeLayout.RIGHT_OF, calloutContent.getId());
+
+              ImageView imageView = new ImageView(getApplicationContext());
+              imageView.setImageDrawable(getResources().getDrawable(R.drawable.ic_info_outline_black_18dp));
```

Comment:
`getDrawable()` is deprecated which will generate a logcat warning, refactor to use the AppCompat library: 

```java
ContextCompat.getDrawable(getActivity(), R.drawable.ic_info_outline_black_18dp);
```